packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -963,12 +963,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         /**
          * Function that return the current server timestamp. This is used by the garbage collector to set the
          * time when a node becomes unreferenced.
-         * For now, we use the timestamp of the last op for gcTimestamp. However, there can be cases where
+         * We use the timestamp of the last op for current timestamp. However, there can be cases where
          * we don't have an op (on demand summaries for instance). In those cases, we will use the timestamp
-         * of this client's connection - https://github.com/microsoft/FluidFramework/issues/8375.
+         * of this client's connection.
          */
-        const getCurrentTimestamp = () => {
-            return this.deltaManager.lastMessage?.timestamp ?? Date.now();
+         const getCurrentTimestamp = () => {
+            const client = this.clientId !== undefined ? this.getAudience().getMember(this.clientId) : undefined;
+            const timestamp = client?.timestamp;
+            return this.deltaManager.lastMessage?.timestamp ?? timestamp ?? Date.now();
```

Comment:
I'd strongly recommend removing Data.now() part and fail fast if we do not have timestamp. This fallback is rather dangerous. I'm sure that in 99.999% of cases you will be fine even with existing code (even before issue https://github.com/microsoft/FluidFramework/issues/7275 is addressed), as we populate audience on "connect" event, long before "connected" for "write" connection fires and triggers summarizer to move forward.
I'd say that even this.deltaManager.lastMessage?.timestamp is not needed, as this code should always run after summarizer connected.

Maybe there are some on-demand cases where it's not the case, but I do not think so.

Reply:
What happens if the container is in storage only mode or other similar scenarios where they may not be an active connection to the server? As I understand, it's not guaranteed that we will always have a delta connection to the server.

`this.deltaManager.lastMessage?.timestamp` is a good optimization because it will give a more recent timestamp than the connected event in majority of the scenarios.

Reply:
I'd think that we should never get here if client is not connected - summarizer does not work in those scenarios.